x-pack/plugins/search_inference_endpoints/server/lib/delete_inference_endpoint.ts

```diff
@@ -5,15 +5,21 @@
  * 2.0.
  */
 
+import { InferenceTaskType } from '@elastic/elasticsearch/lib/api/types';
 import { ElasticsearchClient } from '@kbn/core/server';
 
+function isTaskType(type?: string): type is InferenceTaskType {
+  return type
+    ? ['sparse_embedding', 'text_embedding', 'rerank', 'completion'].includes(type)
+    : true;
+}
```

Comment:
Some questions:
  -  Why do we need to check the validation of task_type? The task type should always be one of those 4 types. 
  - Instead of mentioning the task-types as an array of string, can we gather the types from this [enum](https://github.com/elastic/kibana/blob/7db28682dde859fde8e623a495d267f6e61c2d65/x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/types.ts#L23-L28)? The benefit of this approach is that we do not need to add a new type in multiple places in future.

Reply:
We need to validate it because the client will only accept properly typed strings, and just casting directly is ugly.

I'll use the enum.